src/components/profile/UserInfo.js

```diff
@@ -1,18 +1,51 @@
 import React from "react";
 import SelectedListItem from "../ui/SelectedListItem";
-
-const user = {
-  firstName: "John",
-  middleName: "A.",
-  lastName: "Doe",
-  gender: "Male",
-  dateOfBirth: "1990-05-15",
-  phone: "1234567890",
-  email: "john.doe@example.com",
-  city: "New York",
-  country: "USA",
-};
+import { useEffect, useState } from "react";
+import axios from "axios";
 
 export default function UserInfo() {
-  return <SelectedListItem list={user} />;
+  const token = localStorage.getItem("token");
+  const [userData, setUserData] = useState(null);
+
+  useEffect(() => {
+    const fetchUserData = async () => {
+      if (!token) {
+        console.error("No token found. Please log in.");
+        return;
+      }
+      try {
+        const response = await axios.get("http://localhost:5005/auth/user", {
+          headers: {
+            Authorization: `Bearer ${token}`, // Attach the token to the Authorization header
+            "Content-Type": "application/json",
+          },
+        });
+        const userInfor = {
+          firstName: response.data.firstName,
+          middleName: response.data.middleName,
+          lastName: response.data.lastName,
+          gender: response.data.gender,
+          dateOfBirth: new Date(response.data.dateOfBirth).toLocaleDateString(),
+          phone: response.data.phone,
+          email: response.data.email,
+          city: response.data.city,
+          country: response.data.country,
+        };
+
+        setUserData(userInfor);
+      } catch (error) {
+        console.error(
+          "Error fetching user data:",
+          error.response || error.message
```

Comment:
same right here. And avoid show plain error message to user

Reply:
fixed